### ISSUE TITLE
lxd writing invalid device limits to DB

### ISSUE THREAD

**Issue by nodoubleg** [lxc/lxd#2554]
* Distribution: Ubuntu
 * Distribution version: Xenial
 * The output of "lxc info":

```
apiextensions: []
apistatus: stable
apiversion: "1.0"
auth: trusted
environment:
  addresses:
  - 10.10.10.1:8443
  - 10.10.10.2:8443
  - 10.10.10.3:8443
  architectures:
  - x86_64
  - i686
  driver: lxc
  driverversion: 2.0.5
  kernel: Linux
  kernelarchitecture: x86_64
  kernelversion: 4.4.0-45-generic
  server: lxd
  serverpid: 30512
  serverversion: 2.0.5
  storage: zfs
  storageversion: "5"
config:
  core.https_address: '[::]'
  core.proxy_http: http://1.2.3.4:3128
  core.proxy_https: http://1.2.3.4:3128
  storage.zfs_pool_name: crypt/lxd
public: false
```

# Issue description

When trying to limit a container's disk IO by iops, using the command documented here: https://insights.ubuntu.com/2016/03/30/lxd-2-0-resource-control-412/
```
lxc config device set my-container root limits.read 20Iops
```
results in:
```
error: Invalid integer: 20Iops
```
Interestingly, if you change the number, or the syntax, it still reports the same error.
```
$ lxc config device set stress-ng root limits.read 10 Iops
error: Invalid integer: 20Iops
```
To work around, I have to delete config row from the containers_devices_config table in lxd's sqlite3 DB.

# Steps to reproduce

 1. Create a container
 2. Google for how to limit disk IO by iops, use the procedure listed on the first or second result:```lxc config device set my-container limits.read 20Iops```
 3. get error. User assumes this can be corrected. Adjust syntax in attempt to figure it out:```lxc config device set my-container limits.read 10 Iops```
 4. observe error produced doesn't match command just run.

# Workaround to "unbrick" the container in question.
Just in case somebody else hits this with google and it hasn't been fixed yet:
 1. stop lxd (unsure if necessary) ```sudo systemctl stop lxd```
 2. find row in table containers_devices_config that references the new limits.read or limits.write you attempted to set.```sudo sqlite3 /var/lib/lxd/lxd.db 'SELECT * FROM containers_devices_config WHERE value LIKE "10Iops";```
 3. That should output only one row looking somewhat like this: ```425|304|limits.read|10Iops```
     If not, check the query.
 4. Delete it, adjusting query as necessary: ```sudo sqlite3 /var/lib/lxd/lxd.db 'DELETE FROM containers_devices_config WHERE value LIKE "10Iops";'```

**Comment by stgraber** [comment]
I have already fixed this particular issue, thought I don't believe the fix has made it to any stable release yet.

**Comment by stgraber** [comment]
https://github.com/lxc/lxd/issues/2483

**Comment by giany** [comment]
@stgraber the fix from #2483 is related to memory limit not IOPs..

**Comment by stgraber** [comment]
@giany a commit in the same branch fixes the issue for all other limits by reverting on invalid parse.
